src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -28,6 +31,7 @@ public User register(String email, String password) {
         user.setEmail(email);
         user.setPassword(password);
         userService.add(user);
+        shoppingCartService.registerNewShoppingCart(user);
```

Comment:
check if the user already exists in the DB if not then register a new one or else throw an exception. Create custom RegistrationException and throw